src/errors.rs

```diff
@@ -0,0 +1,2297 @@
+                in this buildpack's repository.
+                !
+                ! This error is almost always a buildpack bug. If you see this error, please file an issue here:
+                ! https://github.com/heroku/buildpacks-debian-packages/issues/new
+            "},
+        );
+    }
+
+    #[test]
+    fn create_package_index_error_create_pgp_verifier() {
+        test_error_output(
+            "
+                Context
+                -------
+                This is a developer error. It should never be seen by an end-user. We validate 
+                release files using PGP certificates from the supported distributions and if there
+                is a problem with how the release files are signed, this error would happen. Our 
+                testing processes should always catch this but, if not, we should direct users to file 
+                an issue.
+            ",
+            CreatePackageIndexError::CreatePgpVerifier(anyhow!("Malformed OpenPGP message")),
+            indoc! {"
+                - Debug Info:
+                  - Malformed OpenPGP message
+
+                ! Failed to verify Release file
+                !
+                ! The PGP signature of the downloaded release file failed verification. This error can \
+                occur if the maintainers of the Debian repository changed the process \
+                for signing release files.
+                !
+                ! Suggestions:
+                ! - Verify if the keys changed by running the ./scripts/extract_keys.sh script \
+                found in this buildpack's repository.
+                !
+                ! This error is almost always a buildpack bug. If you see this error, please file an issue here:
+                ! https://github.com/heroku/buildpacks-debian-packages/issues/new
+            "},
+        );
+    }
+
+    #[test]
+    fn create_package_index_error_write_release_layer() {
+        test_error_output(
+            "
+                Context
+                -------
+                We write downloaded release files and other information into the release layer. I/O 
+                operations can fail for any number of reasons which we can't anticipate and since this
+                file-system area is managed by the buildpack process there is nothing the user can do
+                here other than report it.
+            ",
+            CreatePackageIndexError::WriteReleaseLayer(
+                "/path/to/layer/file".into(),
+                create_io_error("out of memory"),
+            ),
+            indoc! {"
+                - Debug Info:
+                  - out of memory
+
+                ! Failed to write Release file to layer
+                !
+                ! An unexpected I/O error occurred while writing release data to `/path/to/layer/file`.
+                !
+                ! This error is almost always a buildpack bug. If you see this error, please file an issue here:
```

Comment:
I think that we should solve the specific before we solve the general. i.e. we should have a strong proposal and challenge that others provide a better one rather than opening with the general problem and asking for feedback. 

I view this as us doing the work, at least one part of it, right now. 

Regarding CX: I view CX as a useful tool to help our customers and us, but ultimately we own the experience of using buildpack X at Heroku. Both support and ticket volumes are burdens that ultimately fall to the whole languages team (even if there might be a primary engineer that handles them by default). That's my worry here. Both that there will be an impact to the team, and that user will see that message and go into a passive state even in scenarios where they have the power to move the issue forward. Worse than having no actionable suggestion is having a suggestion that doesn't work as expected. Like a "close door" elevator button isn't actually hooked up to anything.

From a values, vision, and methods perspective, I value setting expectations for my customers and stating what is true.

This is the error text:

> This error is almost always a buildpack bug. 
> If you see this error, please file an issue here:

I think that it's not quite true to say it is "almost always" a buildpack bug, because we don't know what triggers that specific code path. If the goal is to learn what could cause that code path, an alternative could be: pre-generate X number of discussions or issues on the repo and emit a unique discussion URL for each of them and prompt people to submit as many details as they possibly can there. 

If we wanted to take that approach we could say:

> The causes for this error are unknown. We do not have suggestions for diagnosis or a workaround at this time. You can help our understanding by sharing your buildpack log and a description of the events in discussion `https://<url-here>`. If you're able to reproduce the problem with an example application and the `pack build` tool, adding that information to the discussion will also help. Once we have more information around the causes of this error we may update this message.

This:

- Tries to state things we know instead of things we guess or assume
- Sets the expectation that we need their help to understand whether or not this is our bug or not
- Directs N discussions to N locations instead of N*P locations (where P is people who encounter the bug).

Though, If we say that, I need us to mean it. I.e. the "We want to know and will update the message with more information" to be true. One thing to consider when making that claim is there are a LOT of these failure cases where we don't know details of why they might happen.

There are 23 instances of this specific error message. If you want I can help open up 23 issues or look into getting discussions enabled in this repo. I still **prefer my original suggestion** to invite people to search through open issues, but I'm trying to understand what you like about this current message and incorporate that feedback.

Reply:
Done. The language I used was the following:

```
The causes for this error are unknown. We do not have suggestions for diagnosis or a 
workaround at this time. You can help our understanding by sharing your buildpack log 
and a description of the issue at:
{issues_url}

If you're able to reproduce the problem with an example application and the {pack} 
build tool ({pack_url}), adding that information to the discussion will also help. Once 
we have more information around the causes of this error we may update this message.
```